### PR TITLE
Adds arbitrary size of integers for varint encoding functions

### PR DESCRIPTION
**Pull request by denizzzka**
(I think it is useful for the future changes)

Also added varint buffer length check to fit into resulting integer value

### CODE REVIEW

**Comment by denizzzka** [comment]
Done!

File: import/dproto/serialize.d [hunk 1]
```diff
@@ -213,21 +228,43 @@ long readVarint(R)(ref R src)
...
-void toVarint(R)(ref R r, ulong src) @property
-	if(isOutputRange!(R, ubyte))
+void toVarint(R, T)(ref R r, T src) @trusted @property
+	if(isOutputRange!(R, ubyte) && isIntegral!T && isUnsigned!T)
 {
-	while(src > 0x7F) {
-		r.put(cast(ubyte)(0x80 | src&0x7F));
+	immutable ubyte maxMask = 0b_1000_0000;
+	
+	while( src >= maxMask )
```
Comment (denizzzka) [inline import/dproto/serialize.d]: sic!

File: import/dproto/buffers.d [hunk 2]
```diff
@@ -108,7 +108,7 @@ struct OptionalBuffer(ulong id, string TypeString, RealType, bool isDeprecated=f
 			} else {
 				CntRange cnt;
 				raw.serializeTo(cnt);
-				toVarint(r, cnt);
+				toVarint(r, cnt.cnt);
```
Comment (msoucy) [inline import/dproto/buffers.d]: Reason for `cnt.cnt` instead of the current `cnt`?

Comment (denizzzka) [inline import/dproto/buffers.d]: Isn't builds current cnt because toVarint() args condition isIntegral!CntRange == false

**Comment by msoucy** [comment]
Other than some weird formatting (the ternary expressions) this looks good to me. Clean up and I'll merge

**Comment by denizzzka** [comment]
> Other than some weird formatting (the ternary expressions) 

In my opinion ternary expressions here well describe the logic of the algorithm.

> Clean up and I'll merge

Format it to your taste - it does not matter for me.

**Comment by denizzzka** [comment]
(ternary expression formatting changed)

**Comment by msoucy** [comment]
Looks good to me - are you comfortable with this being under the boost license? I haven't had  access to a proper computer to change it, but it's planned

**Comment by denizzzka** [comment]
> are you comfortable with this being under the boost license

yes

**Pull request merged**